src/Microsoft.AspNetCore.Razor.Tools/Client.cs

```diff
@@ -58,25 +75,55 @@ internal abstract class Client : IDisposable
             }
         }
 
-        public abstract Stream Stream { get; }
-
-        public void Dispose()
+        private static string GetNextIdentifier()
         {
-            Dispose(disposing: true);
+            var id = Interlocked.Increment(ref counter);
+            return "clientconnection-" + id;
         }
 
-        protected virtual void Dispose(bool disposing)
-        {
-        }
         private class NamedPipeClient : Client
         {
-            public NamedPipeClient(NamedPipeClientStream stream)
+            public NamedPipeClient(NamedPipeClientStream stream, string identifier)
             {
                 Stream = stream;
+                Identifier = identifier;
             }
 
             public override Stream Stream { get; }
 
+            public override string Identifier { get; }
+
+            public async override Task WaitForDisconnectAsync(CancellationToken cancellationToken)
+            {
+                if (!(Stream is PipeStream pipeStream))
```

Comment:
You do this check later as well. When is the Stream not a PipeStream?

Reply:
Yes, We always back it with a PipeStream. But since `Client` is an abstraction and `NamedPipeClient` is only its implementation, I didn't want to assume that it always is a PipeStream.